opentelemetry/src/trace/tracer_provider.rs

```diff
@@ -94,13 +129,42 @@ pub trait TracerProvider {
     /// let tracer = provider.tracer("my_app");
     ///
     /// // tracer used in libraries/crates that optionally includes version and schema url
-    /// let library = std::sync::Arc::new(InstrumentationLibrary::new(
-    ///     env!("CARGO_PKG_NAME"),
-    ///     Some(env!("CARGO_PKG_VERSION")),
-    ///     Some("https://opentelemetry.io/schema/1.0.0"),
-    ///     None,
-    /// ));
+    /// let library = std::sync::Arc::new(
+    ///     InstrumentationLibrary::builder(env!("CARGO_PKG_NAME"))
+    ///         .with_version(env!("CARGO_PKG_VERSION"))
+    ///         .with_schema_url("https://opentelemetry.io/schema/1.0.0")
+    ///         .build(),
+    /// );
+    ///
     /// let tracer = provider.library_tracer(library);
     /// ```
     fn library_tracer(&self, library: Arc<InstrumentationLibrary>) -> Self::Tracer;
 }
+
+#[derive(Debug)]
+pub struct TracerBuilder<'a, T: TracerProvider + ?Sized> {
+    provider: &'a T,
+    library_builder: InstrumentationLibraryBuilder,
+}
+
+impl<'a, T: TracerProvider + ?Sized> TracerBuilder<'a, T> {
+    pub fn with_version(mut self, version: impl Into<Cow<'static, str>>) -> Self {
+        self.library_builder = self.library_builder.with_version(version);
+        self
+    }
+
+    pub fn with_schema_url(mut self, schema_url: impl Into<Cow<'static, str>>) -> Self {
+        self.library_builder = self.library_builder.with_schema_url(schema_url);
+        self
+    }
+
+    pub fn with_attributes(mut self, attributes: impl Into<Vec<KeyValue>>) -> Self {
```

Comment:
To make the API even more flixable we can use something similar to https://github.com/open-telemetry/opentelemetry-rust/blob/e98c61ce73865b56917695dc0ab89ba3eca6c454/opentelemetry/src/trace/tracer.rs#L333-L335